src/msg/async/EventEpoll.cc

```diff
@@ -87,7 +87,7 @@ int EpollDriver::del_event(int fd, int cur_mask, int delmask)
   int mask = cur_mask & (~delmask);
   int r = 0;
 
-  ee.events = 0;
+  ee.events = EPOLLET;
```

Comment:
could change this to 
```c
if (mask) ee.events = EPOLLET;
```
as `&ee` will be ignore by the kernel if we are going to remove the fd from the list being polled.

Reply:
If kernel ignores &ee in case of deletion what is the purpose of the "if (mask)" check?

Reply:
there is chance that the resulting `mask` is not empty -- we are still interested some events not removed by `delmask`, in that case, we will use `EPOLL_CTL_MOD` , and kernel will read `ee`.

Reply:
> As I understood correctly you propose to check mask before setting ee.events, right?

right. 

> That would make sense iif delete path takes care about ee.events value.

agreed. practically, yes!

> my question is what is the purpose of the change you propose?

it's more about readablity -- to emphasis that we don't care about `ee.events`, if we are going to `EPOLL_CTL_DEL` the fd.

Reply:
> it's more about readablity

Readability already sucks, since you set ee.data.fd for deletion as well. So to improve readability something as the following should be done:

```
+++ b/src/msg/async/EventEpoll.cc
@@ -83,16 +83,18 @@ int EpollDriver::del_event(int fd, int cur_mask, int delmask)
 {
   ldout(cct, 20) << __func__ << " del event fd=" << fd << " cur_mask=" << cur_mask
                  << " delmask=" << delmask << " to " << epfd << dendl;
-  struct epoll_event ee;
+  struct epoll_event ee = {0};
   int mask = cur_mask & (~delmask);
   int r = 0;
 
-  ee.events = 0;
-  if (mask & EVENT_READABLE) ee.events |= EPOLLIN;
-  if (mask & EVENT_WRITABLE) ee.events |= EPOLLOUT;
-  ee.data.u64 = 0; /* avoid valgrind warning */
-  ee.data.fd = fd;
   if (mask != EVENT_NONE) {
+    ee.events = EPOLLET;
+    ee.data.fd = fd;
+    if (mask & EVENT_READABLE)
+      ee.events |= EPOLLIN;
+    if (mask & EVENT_WRITABLE)
+      ee.events |= EPOLLOUT;
+
     if ((r = epoll_ctl(epfd, EPOLL_CTL_MOD, fd, &ee)) < 0) {
```

I can do that in the second patch.

Reply:
thanks! `EPOLLET` is already a critical fix IMO.